### DISCUSSION TITLE
Add new semantic token type for an identifier of keyword arguments

### DISCUSSION THREAD

**Discussion by YOYOYO28** [microsoft/pylance-release#2682] [Enhancement]
### Feature request: 
add new semantic token type for an identifier of keyword arguments when calling function.
I'd like to **assign the new semantic token type to ```variable.parameter.function-call.python```**.

### Situation
I use semantic highlighting. Assigning a specific color to ```parameters:python``` allows us to distinguish between newly defined variables in a function and parameters given as function arguments. (before: first picture, after(using semantic highlighting) second picture)

But, with this change, the identifier of the keyword argument when calling a function is also displayed in the same color.
Because the ```variable.parameter.function-call.python```, which represents the identifier in the TextMate rule, was assigned a different color, it was previously displayed in a different color from the other parameters.(In the first picture, ```alpha, beta, gamma``` are displayed in indigo on main method, but second picture, after using semantic highlighting, they are displayed in brown as parameters) This is because semantic highlighting takes precedence over syntax highlighting.

### Suggestion
I propose that assigning the new semantic token type to ```variable.parameter.function-call.python```, such as selfParameter, or clsParameter. 

①
![image](https://user-images.githubusercontent.com/72479755/96407925-6a32d580-121d-11eb-8c86-625bad80fdf0.png)
②
![image](https://user-images.githubusercontent.com/72479755/96407903-5b4c2300-121d-11eb-808c-fd299acd68a0.png)

**Comment by raptor494** [comment]
Would really love if this was added, it's the one thing I miss from before I started using semantic highlighting

**Comment by Atlamillias** [comment]
Would love this as well. Semantic highlighting is one of the few things that keeps me from dropping PyCharm (granted, PyLance has made it MUCH better in VSCode than it was previously in my opinion). 

If not this specific change, then maybe a convenient way of excluding specific scopes from PyLance's semantic highlighting through settings.json so any changes made via "editor.tokenColorCustomizations" will no longer be suppressed? I'm uncertain on how feasible this would be.

**Comment by thehale** [comment]
I'd also really appreciate this feature. To show an additional use case, having colored named parameters would make a huge difference for functions that pass values down a [chain of responsibility](https://refactoring.guru/design-patterns/chain-of-responsibility) and use the same parameter name for clarity.

Additionally, named parameters in decorators currently don't have the semantic type of `parameter`. While this is currently nice because it means my preferred color for named parameters can be applied inside decorators, it would be great if they also got this new semantic type to avoid needing a separate TextMate coloring rule.

![image](https://user-images.githubusercontent.com/47901316/123854894-1e6c1c00-d8d4-11eb-9c3e-e0248a92c8d2.png)
*See this code as a [gist](https://gist.github.com/jhale1805/5f609426cb8f52bd25554086ffd86133)*

As for the specific semantic name, I'd propose `namedParameter` to reflect the naming of `selfParameter` and `clsParameter`. Such a name could also map to similar features in JavaScript and C#. Alternatively, if a more python-specific name was desired, then a semantic name like `keywordArgument` would also work.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by tmke8** [comment]
Just found this issue after searching the internet for the reason for why `variable.parameter.function-call.python` doesn't work. I second the request. I like to see at a glance which variables in a function are parameters to that function. On (neo)vim, I always use this extension to achieve the same effect: https://github.com/m-demare/hlargs.nvim
